client/components/LearnerViewDirective.js

```diff
@@ -477,10 +477,22 @@ tie.directive('learnerView', [function() {
             feedbackDiv.scrollTop = feedbackDiv.scrollHeight +
               additionalHeightForLoadingIndicator;
             $timeout(function() {
-              SolutionHandlerService.processSolutionAsync(
-                tasks[currentTaskIndex], code,
+              var index = 0;
```

Comment:
Interesting -- so you're using sequential execution instead of assembling all the test code and doing a single exec run.

I wonder how the speed of this looks like in practice and how well it scales. If you had a question with 3-4 tasks, how much of a delay do you see in practice when running this on the last task in that question?